test/common/mongo/proxy_test.cc

```diff
@@ -432,10 +434,14 @@ TEST_F(MongoProxyFilterTest, ConcurrentQuery) {
     message->flags(0b11);
     message->cursorId(1);
     message->documents().push_back(Bson::DocumentImpl::create()->addString("hello", "world"));
+    EXPECT_CALL(drain_decision_, drainClose()).WillOnce(Return(true));
```

Comment:
Is it worth it to setup a test where we expect a true drain decision, but there is still an active query?

Reply:
It's impossible. The logic short circuits before drainClose() is called.

Reply:
Sorry, I know the && short circuits. I was more asking if we should also test that this connection drainage doesn't happen with active requests. I guess that is what used to happen before, and previous tests make sure that behavior didnt change.